core/src/saros/session/internal/ActivityHandler.java

```diff
@@ -94,8 +98,14 @@ public void run() {
 
             dispatchQueue.drainTo(pendingActivities);
 
-            for (final List<IActivity> activities : pendingActivities)
+            for (final List<IActivity> activities : pendingActivities) {
+              if (activities == POISON_PILL) {
```

Comment:
As you actually want/need to use the object reference comparison here, how about adding a comment to disable the PMD ([see PMD documentation](https://pmd.github.io/latest/pmd_userdocs_suppressing_warnings.html#nopmd-comment)) check? This would avoid Codacy complaining and would serve as a comment for others reading the code.

```suggestion
              if (activities == POISON_PILL) { // NOPMD - object reference comparison needed
```